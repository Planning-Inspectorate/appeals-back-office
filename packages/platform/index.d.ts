import { HandlerFunction } from 'got';

declare module '@pins/platform' {
	export function createTtlHandler(): HandlerFunction;
	export function validatePostcode(value: string): boolean;
	export function validatePastDate(value: Date | string | number): boolean;
	export function validateFutureDate(value: Date | string | number): boolean;
	export function yesterday(): Date;
}

export interface LocalPlanningDepartment {
	FID: number;
	LPA21CD: string;
	LPA21NM: string;
	Co_terminous: 'Y' | 'N';
}
