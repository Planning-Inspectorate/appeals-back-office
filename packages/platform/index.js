export * from './http/ttl.js';
export * from './util/date.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
