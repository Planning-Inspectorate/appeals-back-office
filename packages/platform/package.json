{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=22.0.0 <= 22.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "test": "cross-env NODE_ENV=test NODE_OPTIONS=\"--experimental-vm-modules --expose-gc\" npx jest --logHeapUsage --ci",
    "test:cov": "cross-env NODE_ENV=test NODE_OPTIONS=\"--experimental-vm-modules --expose-gc\" npx jest --logHeapUsage --ci --coverage",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.3",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.4",
    "@types/lodash-es": "^4.17.12",
    "cross-env": "^7.0.3",
    "jest": "^29.7.0",
    "type-fest": "^4.37.0"
  }
}
