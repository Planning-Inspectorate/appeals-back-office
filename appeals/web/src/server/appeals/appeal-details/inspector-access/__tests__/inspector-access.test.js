import { parseHtml } from '@pins/platform';
import supertest from 'supertest';
import { appealData } from '#testing/app/fixtures/referencedata.js';
import { createTestEnvironment } from '#testing/index.js';
import nock from 'nock';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details';

describe('inspector-access', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /change/:source', () => {
		it('should render changeInspectorAccess page for the lpa answer when source is lpa from appeals details', async () => {
			const appealId = appealData.appealId;
			const response = await request.get(`${baseUrl}/${appealId}/inspector-access/change/lpa`);

			const elementInnerHtml = parseHtml(response.text).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(elementInnerHtml).toContain(
				'Might the inspector need access to the appellant’s land or property?</h1>'
			);
		});

		it('should render changeInspectorAccess page for the appellant answer when source is appellant from appeals details', async () => {
			const appealId = appealData.appealId;
			const response = await request.get(
				`${baseUrl}/${appealId}/inspector-access/change/appellant`
			);

			const elementInnerHtml = parseHtml(response.text).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(elementInnerHtml).toContain(
				'Will an inspector need to access your land or property?</h1>'
			);
		});

		it('should render changeInspectorAccess page for the lpa answer when source is lpa from lpa questionnaire', async () => {
			const appealId = appealData.appealId;
			const lpaQuestionnaireId = appealData.lpaQuestionnaireId;
			const response = await request.get(
				`${baseUrl}/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/inspector-access/change/lpa`
			);

			const elementInnerHtml = parseHtml(response.text).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(elementInnerHtml).toContain(
				'Might the inspector need access to the appellant’s land or property?</h1>'
			);
		});

		it('should render changeInspectorAccess page for the appellant answer when source is appellant from appellant case', async () => {
			const appealId = appealData.appealId;

			const response = await request.get(
				`${baseUrl}/${appealId}/appellant-case/inspector-access/change/appellant`
			);

			const elementInnerHtml = parseHtml(response.text).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(elementInnerHtml).toContain(
				'Will an inspector need to access your land or property?</h1>'
			);
		});
	});

	describe('POST /change/:source', () => {
		it('should re-render changeInspectorAccess page with an error when isRequired is "yes" but details is empty', async () => {
			const appealId = appealData.appealId.toString();

			const invalidData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: ''
			};
			const response = await request
				.post(`${baseUrl}/${appealId}/inspector-access/change/lpa`)
				.send(invalidData);

			expect(response.statusCode).toBe(200);

			const elementInnerHtml = parseHtml(response.text).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(elementInnerHtml).toContain(
				'Might the inspector need access to the appellant’s land or property?</h1>'
			);

			const unprettifiedErrorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(unprettifiedErrorSummaryHtml).toContain('There is a problem</h2>');
			expect(unprettifiedErrorSummaryHtml).toContain('Enter inspector access details</a>');
		});

		it('should re-render changeInspectorAccess page with an error when inspectorAccessDetails exceeds 1000 characters', async () => {
			const appealId = appealData.appealId.toString();
			const invalidData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: 'a'.repeat(1001) // Creates string of 1001 'a' characters
			};

			const response = await request
				.post(`${baseUrl}/${appealId}/inspector-access/change/lpa`)
				.send(invalidData);

			expect(response.statusCode).toBe(200);

			const elementInnerHtml = parseHtml(response.text).innerHTML;
			const unprettifiedErrorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(elementInnerHtml).toMatchSnapshot();
			expect(unprettifiedErrorSummaryHtml).toContain('There is a problem</h2>');
			expect(unprettifiedErrorSummaryHtml).toContain(
				'Inspector access details must be 1000 characters or less</a>'
			);
		});

		it('should re-direct to appeals-details page when data is valid for lpa and came from appeal-details', async () => {
			const appealId = appealData.appealId;
			const lpaQuestionnaireId = appealData.lpaQuestionnaireId;
			const validData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: 'Details'
			};

			nock('http://test/')
				.patch(`/appeals/${appealId}/lpa-questionnaires/${lpaQuestionnaireId}`)
				.reply(200, {
					...validData
				});

			const response = await request
				.post(`${baseUrl}/${appealId}/inspector-access/change/lpa`)
				.send(validData);

			expect(response.statusCode).toBe(302);

			expect(response.text).toBe('Found. Redirecting to /appeals-service/appeal-details/1');
		});

		it('should re-direct to appeals-details page when data is valid for appellant and came from appeal-details', async () => {
			const appealId = appealData.appealId;
			const appellantCaseId = appealData.appellantCaseId;
			const validData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: 'Details'
			};

			nock('http://test/')
				.patch(`/appeals/${appealId}/appellant-cases/${appellantCaseId}`)
				.reply(200, {
					...validData
				});

			const response = await request
				.post(`${baseUrl}/${appealId}/inspector-access/change/appellant`)
				.send(validData);

			expect(response.statusCode).toBe(302);

			expect(response.text).toBe('Found. Redirecting to /appeals-service/appeal-details/1');
		});

		it('should re-direct to appellant case page when data is valid for appellant and came from appellant-case', async () => {
			const appealId = appealData.appealId;
			const appellantCaseId = appealData.appellantCaseId;
			const validData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: 'Details'
			};

			nock('http://test/')
				.patch(`/appeals/${appealId}/appellant-cases/${appellantCaseId}`)
				.reply(200, {
					...validData
				});

			const response = await request
				.post(`${baseUrl}/${appealId}/appellant-case/inspector-access/change/appellant`)
				.send(validData);

			expect(response.statusCode).toBe(302);

			expect(response.text).toBe(
				'Found. Redirecting to /appeals-service/appeal-details/1/appellant-case'
			);
		});

		it('should re-direct to lpaq page when data is valid for lpa and came from lpa-questionnaire', async () => {
			const appealId = appealData.appealId;
			const lpaQuestionnaireId = appealData.lpaQuestionnaireId;
			const validData = {
				inspectorAccessRadio: 'yes',
				inspectorAccessDetails: 'Details'
			};

			nock('http://test/')
				.patch(`/appeals/${appealId}/lpa-questionnaires/${lpaQuestionnaireId}`)
				.reply(200, {
					...validData
				});

			const response = await request
				.post(
					`${baseUrl}/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/inspector-access/change/lpa`
				)
				.send(validData);

			expect(response.statusCode).toBe(302);

			expect(response.text).toBe(
				`Found. Redirecting to /appeals-service/appeal-details/1/lpa-questionnaire/${lpaQuestionnaireId}`
			);
		});
	});
});
