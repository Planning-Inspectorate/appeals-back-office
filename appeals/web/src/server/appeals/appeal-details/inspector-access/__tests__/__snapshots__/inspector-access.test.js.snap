// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inspector-access GET /change/:source should render changeInspectorAccess page for the appellant answer when source is appellant from appeals details 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional govuk-radios__conditional--hidden"
                                    id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3"></textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no" checked>
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`inspector-access GET /change/:source should render changeInspectorAccess page for the appellant answer when source is appellant from appellant case 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional govuk-radios__conditional--hidden"
                                    id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3"></textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no" checked>
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`inspector-access GET /change/:source should render changeInspectorAccess page for the lpa answer when source is lpa from appeals details 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional govuk-radios__conditional--hidden"
                                    id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3"></textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no" checked>
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`inspector-access GET /change/:source should render changeInspectorAccess page for the lpa answer when source is lpa from lpa questionnaire 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional govuk-radios__conditional--hidden"
                                    id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3"></textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no" checked>
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`inspector-access POST /change/:source should re-render changeInspectorAccess page with an error when inspectorAccessDetails exceeds 1000 characters 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#inspector-access-details">Inspector access details must be 1000 characters or less</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" checked data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional" id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3">aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa</textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`inspector-access POST /change/:source should re-render changeInspectorAccess page with an error when isRequired is "yes" but details is empty 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#inspector-access-details">Enter inspector access details</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Will an inspector need to access your land or property?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio" name="inspectorAccessRadio"
                                        type="radio" value="yes" checked data-aria-controls="conditional-inspector-access-radio">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio">Yes</label>
                                    </div>
                                    <div class="govuk-radios__conditional" id="conditional-inspector-access-radio">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label" for="inspector-access-details">Enter reason</label>
                                            <textarea class="govuk-textarea" id="inspector-access-details"
                                            name="inspectorAccessDetails" rows="3"></textarea>
                                        </div>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="inspector-access-radio-2" name="inspectorAccessRadio"
                                        type="radio" value="no">
                                        <label class="govuk-label govuk-radios__label" for="inspector-access-radio-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;
